src/Microsoft.Framework.Caching.SqlServer/SqlQueries.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Caching.SqlServer
+{
+    internal class SqlQueries
+    {
+        private const string CreateTableFormat = "CREATE TABLE [{0}](" +
+            "Id nvarchar(100)  NOT NULL PRIMARY KEY, " +
+            "Value varbinary(MAX) NOT NULL, " +
+            "ExpiresAtTimeUTC datetimeoffset NOT NULL, " +
+            "SlidingExpirationInTicks bigint NULL," +
+            "AbsoluteExpiration datetimeoffset NULL)";
+
+        private const string CreateNonClusteredIndexOnExpirationTimeFormat
+            = "CREATE NONCLUSTERED INDEX Index_ExpiresAtTimeUTC ON [{0}](ExpiresAtTimeUTC)";
+
+        // Used for getting table schema when trying to 'Connect' to the database
+        private const string GetTableSchemaFormat = "SELECT * FROM [{0}]";
+
+        private const string GetCacheItemFormat =
+            "SELECT Id, Value, ExpiresAtTimeUTC, SlidingExpirationInTicks, AbsoluteExpiration " +
+            "FROM [{0}] WHERE Id = @Id AND @UtcNow <= ExpiresAtTimeUTC";
```

Comment:
String equality in SQL Server will have different semantics to CLR, e.g. depending on the database collation it can be case insensitive and accent sensitive, etc. It will also ignor trailing blanks. Assuming you want to have invariant culture semantics on database lookups and that you want to make this really robust, you have several (non mutually exclusive) options:
1. Specify a collation that satisfies your need on this column on table creation.
2. Always trim trailling spaces or otherwise this if any is detected everywhere the id is taken as an argument.
3. Always encode the string as a byte[]
4. Check again in memory for equality after you got the value (note that this will only help with lookups).